### DISCUSSION TITLE
project dead?

### DISCUSSION THREAD

**Discussion by JustinBondSF** [203-Systems/MatrixOS#39] [General]
Is this project dead? I bought this thing hoping that the open source community would keep bringing new feautures, but it doesnt really seem as though theres even any documentation on developing applications for the device, which is a shame.  As it stands now, the limited functionality and DAW integration (or lack of) severely handicap the usefulness of the device, and i always find myself reaching for my launchpad instead.

**Comment by DRCRecoveryData** [comment]
> Is this project dead? I bought this thing hoping that the open source community would keep bringing new feautures, but it doesnt really seem as though theres even any documentation on developing applications for the device, which is a shame.  As it stands now, the limited functionality and DAW integration (or lack of) severely handicap the usefulness of the device, and i always find myself reaching for my launchpad instead.

As null said he busy this time, also he stransfering to new library for development new firmware

**Comment by 203Null** [comment]
Nan. Been working in lower level USB stuffs in my private branch. Decided to move up the major feature of Matrix 3.0 up, dynamic USB reconfiguration because I want to do custom map right.

Stacked a few feature branch but still working on the low level rework to be done and stabilize the new reworked APIs.


It also a challenging year for me so dev works has been on and off. I'm truly sorry for that.

![image](https://github.com/203Electronics/MatrixOS/assets/19933093/d87ea3f0-4057-434c-b38c-89cbe2a63377)

**Reply by 203Null** [reply]
Also if you have some request. Let me know. I do wish there are better DAW integration but I'm not really a DAW wizard, especially with anything not Ableton. I hope the community can fill those voids.

**Reply by nvmker** [reply]
About DAW integration, LP X and Pro MK3 have a mode in which it can be put by sending I guess sysex or like that to a particular MIDI interface of LP, after that software developers can control leds and so forth and use it with their integration with software, it's all described in LP programmer reference guide, and it's used to control session mode for Live/Logic and also in some other software. It will be useful to have such mode for Matrix, so devs can integrate it in a software.

**Reply by 203Null** [reply]
Do you mean those sysex adding fader etc on the pad? I'm currently approaching this via custom map but not via any sysex or other protocol. I will see what I can do about that

**Reply by nvmker** [reply]
Sysex putting LP in a programmer mode where developers can control LEDs and do other custom stuff via software, you can check API implementation in a programmer reference manual in LP X/Pro MK3.
